src/main.rs

```diff
@@ -272,6 +274,57 @@ fn main() {
         }
     };
 
+    let custom_pages_dir = config
+        .directories
+        .custom_pages_dir
+        .as_ref()
+        .map(PathWithSource::path);
+
+    // Note: According to the TLDR client spec, page names must be transparently
+    // lowercased before lookup:
+    // https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md#page-names
+    let command = args.command.join("-").to_lowercase();
+
+    if args.edit_patch || args.edit_page {
+        let Some(custom_pages_dir) = custom_pages_dir else {
+            print_error(enable_styles, &anyhow!("Fail to get custom page dir"));
+            process::exit(1);
+        };
+        let _ = create_dir_all(custom_pages_dir);
```

Comment:
Why are we ignoring errors here? If this fails, the editing with `EDITOR` might not work afterwards, or am I not getting something?

Reply:
Maybe custom_pages_dir exists already

Reply:
I thought that this wouldn't error - let's add a test for this and then see if checking for errors here still works

Reply:
https://doc.rust-lang.org/std/fs/fn.create_dir_all.html

Reply:
Yes, I read that too, but it doesn't support your point. In fact, running a fresh project with just
```rust
fn main() {
    std::fs::create_dir_all("test/dir").unwrap();
    std::fs::create_dir_all("test/dir").unwrap();
}
```
still works - the function will not error if (part of) the path already exists.

Reply:
Maybe on some platforms it will return err.
Anyway, if an io error happens, `EDITOR` will catch it too

Reply:
No, we shouldn't rely on the editor behavior. In fact, for example neovim happily opens non-existent and impossible files, for example I can run `nvim Cargo.toml/foo` and will get an editor as usual.

I stand by my point, we should have tests to ensure that tealdeer still works if the directories exist. Once we have those, I think we will see that using `create_dir_all(...)?` should do what we want

Reply:
> nvim Cargo.toml/foo

 Then you can't save it

Reply:
There is a check at L298-305

Reply:
Yes, and by this, the experience is more frustrating than it has to be - let's not spawn the editor in this case 

Please add a test for when the directory already exists, and one for when there is a file where the directory should be, then we can check which implementation fulfills the requirements